rfcs/react-components/non-core-components.md

```diff
@@ -0,0 +1,388 @@
+- 👍 Easy to update when using multiple components
+
+##### Cons
+
+- 👎 Needs to follow the breaking change cadence of each components
+- 👎 Slows down breaking changes for all components
+
+#### Individual packages
+
+Users should install each component package individually.
+
+##### Pros
+
+- 👍 Easy to pick and choose components that might not always be relevant
+- 👍 Allows more flexible breaking change cadence
+
+##### Cons
+
+- 👎 Easier to install duplicate versions
+- 👎 We might be encouraged to break too often
+
+### How to depend on core components
+
+#### Depend directly on components packages
+
+Each non-core component should depend on the individual core components
+
+##### Pros
+
+- 👍 Users will always install the required core components
+
+##### Cons
+
+- 👎 Consumes core packages in a way we don't recommend other users
+- 👎 Dependency graph could be hard to manage/track
+- 👎 Internal API might become public
+
+#### Depend on suite package is peer dependency
+
+#### Pros
+
+- 👍 No unnecessary dependencies - we expect users to have core suite package installed
+- 👍 Other examples in OSS - Material UI/React all recommend peer dependencies
+- 👍 Easy for users to consume as long as core does not break
+- 👍 We test how 'complete' our public API is
+
+#### Cons
+
+- 👎 V10 upgrade could be unpleasant - need to check what range is supported for each package
+- 👎 More of our internal API becomes public
+
+### Documentation
+
+Fluent UI production components should always be documented regardless of whether they are core/non-core components.
+This section describes the ways to document non-core components.
+
+#### Current docsite
+
+Create a new section in our docsite explicitly for non-core components alongside core components.
+
+##### Pros
+
+- 👍 Existing infrastructure already setup
+- 👍 Enforces consistent documentation approach
+
+##### Cons
+
+- 👎 Do we want to document non-core components in the same way?
+- 👎 For separate repo, would need to think about how to consume stories
+- 👎 Needs work to support independently versioned packages
+
+#### New docsite
+
+Creates a new docsite for non-core components that will be linked to the current docsite.
+
+##### Pros
+
+- 👍 It's not hard to link between docsites that share the same look and feel
+- 👍 Clear boundary for users to distinguish core/non-core components
+- 👍 Possible to change documentation approach
+
+##### Cons
+
+- 👎 Manage the infra for another domain and publish
+- 👎 Needs work to support independently versioned packages
+
+### Parnter ownership and collaboration
+
+Partner teams might want to build scenario/product specific components, with guidance and support from the Fluent UI
+team. The Fluent UI team should have a say in the quality of the code that is shipped by partner teams under the
+Fluent UI umbrella and work with them to establish these quality criteria and ownership. This section proposes
+two different ways we should host partner code.
+
+#### GitHub
```

Comment:
while I understand that products live within ADO and their security needs etc, this is the ideal/most standard scenario. Any other approach you suggest will complicate infra to huge extent with what benefit ?...

To add,  those extension cannot ever contain business related logic so the security concern should is non present ?

 it's the main purpose of OSS extensions to be and live in the "open"

Reply:
Resolving this comment in favour of the one on private packages, IMO that's the deciding factor on if we want any kind of ADO support